src/script/lua_api/l_util.cpp

```diff
@@ -307,9 +347,16 @@ int ModApiUtil::l_decompress(lua_State *L)
 	size_t size;
 	const char *data = luaL_checklstring(L, 1, &size);
 
+	LuaCompressMethod method = get_compress_method(L, 2);
+
 	std::istringstream is(std::string(data, size), std::ios_base::binary);
 	std::ostringstream os(std::ios_base::binary);
-	decompressZlib(is, os);
+
+	if (method == LUA_COMPRESS_METHOD_DEFLATE) {
+		decompressZlib(is, os);
+	} else if (method == LUA_COMPRESS_METHOD_ZSTD) {
+		decompressZstd(is, os);
+	}
```

Comment:
Same here: should probably return `nil` to indicate failure in the else-case.

Reply:
See comment to previous review comment.